main.tf

```diff
@@ -59,61 +52,44 @@ resource "aws_route_table" "public" {
   }
 }
 
-resource "aws_route_table_association" "public1" {
-  subnet_id      = aws_subnet.public1.id
-  route_table_id = aws_route_table.public.id
-}
+resource "aws_route_table_association" "public" {
+  count = length(local.public_cidr)
 
-resource "aws_route_table_association" "public2" {
-  subnet_id      = aws_subnet.public2.id
+  subnet_id      = aws_subnet.public[count.index].id
   route_table_id = aws_route_table.public.id
 }
 
-resource "aws_eip" "nat1" {
-  vpc      = true
-}
+resource "aws_eip" "nat" {
+  count = length(local.public_cidr)
 
-resource "aws_eip" "nat2" {
-  vpc      = true
+  vpc = true
 }
 
-resource "aws_nat_gateway" "nat1" {
-  allocation_id = aws_eip.nat1.id
-  subnet_id     = aws_subnet.public1.id
-}
+resource "aws_nat_gateway" "nat" {
+  count = length(local.public_cidr) 
 
-resource "aws_nat_gateway" "nat2" {
-  allocation_id = aws_eip.nat2.id
-  subnet_id     = aws_subnet.public2.id
+  allocation_id = aws_eip.nat[count.index].id
+  subnet_id = aws_subnet.public[count.index].id
 }
 
-resource "aws_route_table" "private1" {
+resource "aws_route_table" "private" {
+  count = length(local.public_cidr)
```

Comment:
it is ideal you use `count = length(local.private_cidr)` for resources related to the private subnet